File: frontend/src/types/carousel.ts
export interface CarouselItem {
  id: number
  image: string
  title?: string
  subtitle?: string
  link?: string
  buttonText?: string
  order: number
  enabled: boolean
  createdAt: string
  updatedAt: string
}

export interface CarouselCreateRequest {
  image: string
  title?: string
  subtitle?: string
  link?: string
  buttonText?: string
  order?: number
  enabled?: boolean
}

export interface CarouselUpdateRequest {
  image?: string
  title?: string
  subtitle?: string
  link?: string
  buttonText?: string
  order?: number
  enabled?: boolean
}

File: backend/src/main/java/com/example/culturalxinjiang/entity/Carousel.java
package com.example.culturalxinjiang.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "carousels")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Carousel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String image;

    private String title;

    private String subtitle;

    private String link;

    private String buttonText;

    @Column(name = "display_order", nullable = false)
    @Builder.Default
    private Integer order = 0;

    @Column(nullable = false)
    @Builder.Default
    private Boolean enabled = true;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(nullable = false)
    private LocalDateTime updatedAt;
}

